docs: update `no-undef-init` when not to use section (#19624)
eslint/eslint: docs/src/rules/no-undef-init.md

## When Not To Use It

> There are two situations where initializing to `undefined` behaves differently than omitting the initialization.
> 
> The first is when a `var` declaration occurs inside of a loop. For example:

Reviewer: Are we sure there are only two situations?
```suggestion
There are situations where initializing to `undefined` behaves differently than omitting the initialization. 

One such case is when a `var` declaration occurs inside a loop. For example:
```

```

:::

Secondly, this behavior can also be observed when a variable is redeclared using `var`:

```js
function foo() {
    var x = 1;
    console.log(x); // output: 1
    
    var x;
    console.log(x); // output: 1

    var x = undefined; // output: undefined
    console.log(x);
}

foo();

> ```

Reviewer 2: ````suggestion
```

In this case, you can avoid redeclaration by changing it to an assignment (`x = undefined;`), or use an eslint disable comment on a specific line.
````

Author: Done!

```

:::

> The second is when a variable is redeclared using `var`. For example:

Reviewer: ```suggestion
Another such case is when a variable is redeclared using `var`. For example:
```

Author: Done!
